.setup/bin/recreate_submissions.py

```diff
@@ -0,0 +1,294 @@
+    def update_egver(g_id: str, user_id: str, active_version: int):
+        if ARG_VERBOSE:
+            print(f'    update_egver()')
+            print(f'    g_id           = {g_id}')
+            print(f'    user_id        = {user_id}')
+            print(f'    active_version = {active_version}')
+        query = table_egver.select().where(
+            (table_egver.c.g_id == g_id) &
+            (table_egver.c.user_id == user_id)
+        )
+        ret = dbconn_course.execute(query)
+        rows = ret.all()
+        if ARG_VERBOSE:
+            print(f'    SQL: {query}')
+            print(f'    egver_data = {rows}')
+        if len(rows) == 0: # INSERT
+            query = table_egver.insert().values(
+                g_id = g_id,
+                user_id = user_id,
+                active_version = active_version
+            )
+        else: # UPDATE
+            query = table_egver.update().values(
+                active_version = active_version
+            ).where(
+                (table_egver.columns.g_id == g_id) &
+                (table_egver.columns.user_id == user_id)
+            )
+        if ARG_VERBOSE:
+            print(f'    SQL: {query}')
+        dbconn_course.execute(query)
+    for user in os.listdir(ARG_SUBMISSIONS):
+        if not os.path.isdir(os.path.join(ARG_SUBMISSIONS,user)):
+            continue
+        if user not in user_list:
+            print(f'WARNING: found dir for {user} who is not in the course')
+            continue
+        print(f'SUBMITTING FOR USER {user}')
+        user_dirlist = os.listdir(os.path.join(ARG_SUBMISSIONS,user))
+        versions = [int(f) for f in user_dirlist if os.path.isdir(
+            os.path.join(ARG_SUBMISSIONS,user,f))]
+        # determine active version
+        if os.path.isfile(os.path.join(ARG_SUBMISSIONS,user,UASF_NAME)):
+            with open(os.path.join(ARG_SUBMISSIONS,user,UASF_NAME)) as uasfile:
+                uasdata = json.load(uasfile)
+            active_version = uasdata['active_version']
+        else: # use highest
+            active_version = max(versions)
+        for version in sorted(versions):
+            # skip inactive versions with active only flag
+            if ARG_ACTIVE_ONLY and version != active_version:
+                continue
+            print(f'  SUBMITTING VERSION {version}')
+            submit(ARG_SEMESTER,ARG_COURSE,ARG_GRADEABLE,user,
+                os.path.join(ARG_SUBMISSIONS,user,str(version)),
+                insert_egdata,update_egver)
+    dbconn_course.close()
+    # initiate batch regrade
+    print(f'ADDING SUBMISSIONS TO GRADING QUEUE')
+    regrade_py = os.path.join(ARG_INSTALL_DIR,'bin','regrade.py')
+    gradeable_path = os.path.join(ARG_DATA_DIR,'courses',
+        ARG_SEMESTER,ARG_COURSE,'submissions',ARG_GRADEABLE)
+    os.system(f'{regrade_py} {gradeable_path}')
+
+if __name__ == '__main__':
+    if getpass.getuser() != 'root':
+        print('Please run this as root, exiting...')
+        quit()
```

Comment:
Why not restructure this to only call `main()` if the user has root access instead of exiting if the user doesn't?